databases/de-database-schema/src/main/functions/02_app_count.sql

```diff
@@ -31,3 +31,23 @@ CREATE OR REPLACE FUNCTION app_count(uuid, boolean) RETURNS bigint AS $$
     END
 $$ LANGUAGE SQL;
 
+--
+-- Yet another version of the same function that allows us to limit the
+-- count to certain app IDs.
+--
+CREATE OR REPLACE FUNCTION app_count(uuid, boolean, anyarray) RETURNS bigint AS $$
+    SELECT COUNT(DISTINCT a.id) FROM apps a
+    JOIN app_category_app aca ON a.id = aca.app_id
+    WHERE NOT a.deleted
+    AND aca.app_category_id IN (SELECT * FROM app_category_hierarchy_ids($1))
+    AND a.id = ANY ($3::uuid[])
+    AND CASE
+        WHEN $2 THEN TRUE
+        ELSE NOT EXISTS (
+            SELECT * FROM app_steps s
+            JOIN tasks t ON t.id = s.task_id
+            WHERE s.app_id = a.id
+            AND t.external_app_id IS NOT NULL
+        )
+    END
+$$ LANGUAGE SQL;
```

Comment:
The third parameter is used by the app listing services to list only the apps that the user has permission to view.